components/skus/browser/skus_service_impl.cc

```diff
@@ -109,9 +114,30 @@ void SkusServiceImpl::RefreshOrder(
     mojom::SkusService::RefreshOrderCallback callback) {
   std::unique_ptr<skus::RefreshOrderCallbackState> cbs(
       new skus::RefreshOrderCallbackState);
-  cbs->cb = std::move(callback);
-  GetOrCreateSDK(domain)->refresh_order(OnRefreshOrder, std::move(cbs),
-                                        order_id);
+  cbs->cb = base::BindOnce(&SkusServiceImpl::OnRefreshOrder,
+                           weak_factory_.GetWeakPtr(), std::move(callback));
+
+  sdk_task_runner_->PostTask(
+      FROM_HERE, base::BindOnce(&SkusServiceImpl::RefreshOrderTask,
```

Comment:
you should almost never call the same class on more than one thread

Reply:
there may have been some recent changes in WeakPtr that make this safe, checking into it

Reply:
nope, as I originally thought this is not safe because SkusServiceImpl can be destroyed while a task is running on sdk task runner